nomos-services/tx-service/src/tx/service.rs

```diff
@@ -211,7 +213,7 @@ where
     Pool: MemPoolTrait<Storage = StorageAdapter> + RecoverableMempool + Send + Sync,
     StorageAdapter: MempoolStorageAdapter<RuntimeServiceId> + Clone + Send + Sync,
     <Pool as RecoverableMempool>::RecoveryState: Debug + Send + Sync,
-    Pool::Item: Clone + Send + 'static,
+    Pool::Item: Transaction<Hash = Pool::Key> + Clone + Send + 'static,
```

Comment:
At this level we already know that the pool item is `Transaction`, so the bind to transaction should be in this service instead of the pool.